### ISSUE TITLE
parsing ambiguities for set brackets

### ISSUE THREAD

**Issue by daejunpark** [kframework/k-legacy#1931]
```
<busy> Busy (.Set => SetItem(V)) </busy>
```

```
$ kompile kool-untyped.k
[Error] Compiler: Had 1 parsing errors.
[Error] Critical: Unexpected sort Set for term Busy. Expected KLabel.
    Source(./kool-untyped.k)
    Location(538,15,538,19)
```

on the top of ba05f53c83dedde7ac7351ad34958acad2c02664

**Comment by cos** [comment]
@radumereuta, I guess `Busy` is considered a klabel and the `KApply` production is preferred, right?

**Comment by grosu** [comment]
Welcome back to our good old friend, the parentheses.

@daejunpark , I'm assuming you are talking about this old rule, right?

```
 rule <k> acquire V:Val; => .K ...</k>
       <holds>... .Map => V |-> 0 ...</holds>
       <busy> Busy `.Set => SetItem(V)` </busy>
    when (notBool(V in Busy:Set))
```

If yes, then I believe this is a parsing bug, right @radumereuta?

The K syntax should have priority when there are ambiguities.  But it should not disallow other parses which are possible otherwise.  In the rule above, we know that `Busy` has sort `Set` (user-declared in the side condition), so there should be no ambiguity on how to parse the contents of the `<busy/>` cell.

**Comment by daejunpark** [comment]
@grosu Yes, this issue came from that rule.

**Comment by daejunpark** [comment]
fixed via #2048
